src/System.Private.ServiceModel/tools/IISHostedWcfService/App_code/testhosts/BasicHttpXmlSerFormatAttrServiceHost.cs

```diff
@@ -0,0 +1,162 @@
+        {
+            var serviceHost = new RpcEncSingleNsServiceHost(serviceType, baseAddresses);
+            return serviceHost;
+        }
+    }
+
+    public class RpcEncSingleNsServiceHost : TestServiceHostBase<ICalculator>
+    {
+        protected override string Address { get { return "Basic"; } }
+
+        protected override Binding GetBinding()
+        {
+            return new BasicHttpBinding();
+        }
+
+        public RpcEncSingleNsServiceHost(Type serviceType, params Uri[] baseAddresses)
+            : base(serviceType, baseAddresses)
+        {
+            //multiple binding namespaces test: need to figure out if to add new servicehost for it, code would be like below.
+            //Binding binding = GetBinding();
+            //binding.Namespace = "http://some.Namespace.org/";
+            //ServiceEndpoint endpoint2 = this.AddServiceEndpoint(typeof(ICalculator), binding, Address+"/extra");
+        }
+    }
+
+    public class RpcLitSingleNsServiceHostFactory : ServiceHostFactory
+    {
+        protected override ServiceHost CreateServiceHost(Type serviceType, Uri[] baseAddresses)
+        {
+            var serviceHost = new RpcLitSingleNsServiceHost(serviceType, baseAddresses);
+            return serviceHost;
+        }
+    }
+
+    public class RpcLitSingleNsServiceHost : TestServiceHostBase<ICalculator>
+    {
+        protected override string Address { get { return "Basic"; } }
+
+        protected override Binding GetBinding()
+        {
+            return new BasicHttpBinding();
+        }
+
+        public RpcLitSingleNsServiceHost(Type serviceType, params Uri[] baseAddresses)
+            : base(serviceType, baseAddresses)
+        {
+        }
+    }
+
+    public class DocLitSingleNsServiceHostFactory : ServiceHostFactory
+    {
+        protected override ServiceHost CreateServiceHost(Type serviceType, Uri[] baseAddresses)
+        {
+            var serviceHost = new DocLitSingleNsServiceHost(serviceType, baseAddresses);
+            return serviceHost;
+        }
+    }
+
+    public class DocLitSingleNsServiceHost : TestServiceHostBase<ICalculator>
+    {
+        protected override string Address { get { return "Basic"; } }
+
+        protected override Binding GetBinding()
+        {
+            return new BasicHttpBinding();
+        }
+
+        public DocLitSingleNsServiceHost(Type serviceType, params Uri[] baseAddresses)
+            : base(serviceType, baseAddresses)
+        {
+        }
+    }
+
+    public class RpcEncMultiNsServiceHostFactory : ServiceHostFactory
+    {
+        protected override ServiceHost CreateServiceHost(Type serviceType, Uri[] baseAddresses)
+        {
+            var serviceHost = new RpcEncMultiNsServiceHost(serviceType, baseAddresses);
+            return serviceHost;
+        }
+    }
+
+    public class RpcEncMultiNsServiceHost : TestServiceHostBase<ICalculator, IHelloWorld>
```

Comment:
Can we rename this to `XmlSerializerDualContractServiceHost` and reuse it for all service implementing `TestServiceHostBase<ICalculator, IHelloWorld>` at https://github.com/dotnet/wcf/pull/2341/files#diff-7befb2b82d7343194e5888fe3775a510R50?

Reply:
Updated.